src/Nethermind/Nethermind.Consensus.AuRa/InitializationSteps/StartBlockProducerAuRa.cs

```diff
@@ -224,9 +224,7 @@ private BlockProducerEnv GetProducerChain(ITxSource? additionalTxSource)
         BlockProducerEnv Create()
         {
             ReadOnlyBlockTree readOnlyBlockTree = _api.BlockTree.AsReadOnly();
-
-            ReadOnlyTxProcessingEnv txProcessingEnv = _api.CreateReadOnlyTransactionProcessorSource();
-            IReadOnlyTxProcessingScope scope = txProcessingEnv.Build(Keccak.EmptyTreeHash);
+            IReadOnlyTxProcessingScope scope = _api.ReadOnlyTxProcessorSource.Build(Keccak.EmptyTreeHash);
```

Comment:
not disposing the scope? Maybe add it to dispose stack?

Reply:
Congratulation! You've found a long lived scope. At least its explicit now.